### PR TITLE
fix(Core/Chat): Channel exploit

### PR DESCRIPTION
**Pull request by wizzymore**
##### CHANGES PROPOSED:
-  Checking if the owner of the channel is a GM and is invisible

##### ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2155

##### TESTS PERFORMED:
- Tested by using 2 accounts, gm one and player one

##### HOW TO TEST THE CHANGES:

Using the gm account join a channel:
/join CHANNEL_NAME
Using the player account join the same channel:
/join CHANNEL_NAME
Leave the channel with the GM account, it shouldn't say nothing.
Do the same order of joining a channel but leave with the player first ( who should now be the owner even tho he joined second ) and you will see a text that the GM ( the only player in the channel ) is the owner.

##### Target branch(es):
- [x] Master

##### Operating System
 - [x] Windows
 - [x] Linux

 

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by pklloveyou** [comment]
Tested and working perfectly

**Comment by wizzymore** [comment]
Yes @Stoabrogga you are right, we need to check based on that config.

I will make the change tomorrow or on monday as i am out of town.

**Comment by Stoabrogga** [comment]
Great, thanks. No need to hurry, take your time.

**Comment by wizzymore** [comment]
I think this should do it

**Comment by wizzymore** [comment]
I will try your method and follow up with feedback if any

**Comment by wizzymore** [comment]
Your method is just announcing when the owner is changes, we don't want that, we wan't to change the owner when the owner is a gm and the config GM Silent Join is enabled to the player that entered the channel, and we want to do this silently as we don't want to announce the player that he is the owner, that meaning either someone just left the channel or someone else is on the channel

**Comment by Stoabrogga** [comment]
Ok, thanks for the explanation, I'll look into this again later.

File: src/server/game/Chat/Channels/Channel.h [hunk 1]
```diff
@@ -197,6 +201,7 @@ class Channel
         void UnBan(uint64 guid);
         void Password(Player const* player, std::string const& pass);
         void SetMode(Player const* player, std::string const& p2n, bool mod, bool set);
+        void SetOwnerGM(Player const* player, bool on);
```
Comment (Stoabrogga) [inline src/server/game/Chat/Channels/Channel.h]: This function is not used, please remove.

Comment (wizzymore) [reply src/server/game/Chat/Channels/Channel.h]: Why not leaving there as an extension of the function for future edits?
I don't think is does anything wrong

If you really want to delete it i will

**Comment by wizzymore** [comment]
Now this is working as inteded, thank your for all the help @Stoabrogga !

File: src/server/game/Chat/Channels/Channel.h (hunk 1 above)
Comment (FrancescoBorzi) [reply src/server/game/Chat/Channels/Channel.h]: I would avoid adding unused code

Comment (Stoabrogga) [reply src/server/game/Chat/Channels/Channel.h]: @wizzymore 
Please remove the function. It is generally bad practice to leave dead code in software projects. It just increases the complexity of the code and makes it harder to maintain the software.

**Comment by wizzymore** [comment]
Please label this as Needs to be tested so Testers can look into the PR

**Comment by Stoabrogga** [comment]
Done. I'll also test this later once more.

**Comment by wizzymore** [comment]
Ty for testing @Stoabrogga 

Idk why i get so many white spaces problems in Sublime Text.. I think i really need to change to VS Code

**Comment by FrancescoBorzi** [comment]
@wizzymore just use CLion since you have the license already ;)

**Pull request merged**